eng/versioning/pom_file_version_scanner.ps1

```diff
@@ -0,0 +1,370 @@
+                    $retVal = Test-Dependency-Tag-And-Version $libHash $extDepHash $versionNode.InnerText $versionNode.NextSibling.Value
+                    if ($retVal)
+                    {
+                        $script:FoundError = $true
+                        Write-Output "$($retVal)"
+                    }
+                }
+            }
+            else 
+            {   
+                $script:FoundError = $true
+                # every project string needs to have an update tag and projects version tags are always 'current'
+                Write-Output "Error: Missing project/parent/version update tag. The tag should be <!-- {x-version-update;$($groupId):$($artifactId);current} -->"
+            }
+        }
+        else 
+        {
+            # output an error for missing version element
+            $script:FoundError = $true
+            Write-Output "Error: Could not find project/parent/version node for $($pomFile)"
+        }    
+    }
+
+    # Verify every dependency as a group, artifact and version
+    # GetElementsByTagName should get all dependencies including dependencies under plugins
+    foreach($dependencyNode in $xmlPomFile.GetElementsByTagName("dependency"))
+    {
+        $artifactId = $dependencyNode.artifactId
+        $groupId = $dependencyNode.groupId
+        $versionNode = $dependencyNode.GetElementsByTagName("version")[0]
+        if ($versionNode.NextSibling -and $versionNode.NextSibling.NodeType -eq "Comment") 
+        {
+            # unfortunately because there are POM exceptions we need to wildcard the group which may be 
+            # something like <area>_groupId
+            if (!$versionNode.NextSibling.Value.Trim() -like ("{x-version-update;*$($groupId):$($artifactId);"))
+            {
+                $script:FoundError = $true
+                Write-Output "Error: dependency version update tag for groupId=$($groupId), artifactId=$($artifactId) should be <!-- {x-version-update;$($groupId):$($artifactId);current|dependency|external_dependency<select one>} -->"
+            }
+            else
+            {
+                # verify the version tag and version are correct
+                $retVal = Test-Dependency-Tag-And-Version $libHash $extDepHash $versionNode.InnerText $versionNode.NextSibling.Value
+                if ($retVal)
+                {
+                    $script:FoundError = $true
+                    Write-Output "$($retVal)"
+                }
+            }
+        }
+        else 
+        {   
+            $script:FoundError = $true
+            # <!-- {x-version-update;<groupId>:<artifactId>;current} -->
+            Write-Output "Error: Missing dependency version update tag for groupId=$($groupId), artifactId=$($artifactId). The tag should be <!-- {x-version-update;$($groupId):$($artifactId);current|dependency|external_dependency<select one>} -->"
+        }
+    }
+    # Verify every plugin as a group, artifact and version
```

Comment:
```suggestion
    # Verify every plugin has a group, artifact and version
```

Reply:
Fixed locally, will be updated with the next commit.